apps/jubmoji-quest/src/hooks/useFetchPowers.ts

```diff
@@ -0,0 +1,95 @@
+import {
+  createJubmojiPowerProof,
+  verifyJubmojiPowerProof,
+} from "@/lib/proving";
+import { JubmojiPower } from "@/types";
+import { Jubmoji } from "jubmoji-api";
+import { useMutation, useQuery } from "react-query";
+
+interface PowerMutationProps {
+  power: JubmojiPower;
+  jubmojis: Jubmoji[];
+}
+
+export const useFetchPowers = () => {
+  return useQuery(
+    ["powers"],
+    async (): Promise<JubmojiPower[]> => {
+      const response = await fetch("/api/powers");
+
+      if (!response.ok) {
+        console.error("Could not fetch powers.");
+        return [];
+      }
+
+      const powers: JubmojiPower[] = await response.json();
+
+      return powers;
+    },
+    {
+      refetchOnWindowFocus: false,
+    }
+  );
+};
+
+export const useFetchPowerById = (id: string | string[] | undefined) => {
+  return useQuery(
+    ["powers", id],
+    async (): Promise<JubmojiPower | null> => {
+      const response = await fetch(`/api/powers/${id}`);
+
+      if (!response.ok) {
+        console.error(`Could not fetch power ${id}.`);
+        return null;
+      }
+
+      const power: JubmojiPower = await response.json();
+
+      return power;
+    },
+    {
+      refetchOnWindowFocus: false,
+    }
+  );
+};
+
+/**
+ * Execute power
+ * @returns
+ */
+export const usePowerMutation = () => {
+  return useMutation({
+    mutationKey: "usePower",
+    mutationFn: async ({ power, jubmojis }: PowerMutationProps) => {
+      let serializedProof;
+      try {
+        serializedProof = await createJubmojiPowerProof(power, jubmojis);
+      } catch (error) {
+        return { error: "Failed to use your power!" };
+      }
+
+      let { verified } = await verifyJubmojiPowerProof(power, serializedProof);
```

Comment:
We can delete verification here, its just takes time and we do it on the server side anyways.